org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMElementFormatter.java

```diff
@@ -124,7 +124,17 @@ private int formatStartTagElement(DOMElement element, XMLFormattingConstraints p
 				if (element.isSelfClosed()) {
 					// <foo/> --> <foo />
 					int offset = element.getEnd() - 2;
-					if (isSpaceBeforeEmptyCloseTag()) {
+					boolean isSingleAttribute = element.getAttributeNodes() != null
+							? element.getAttributeNodes().size() == 1
+							: true;
+					if (isClosingBracketNewLine() && isSplitAttributes() && !isSingleAttribute) {

```

Comment:
I will create a private method which embeds this code because https://github.com/eclipse/lemminx/pull/1274/files#diff-7f29cacc7a55ad6a31c48e92878f142024e9c20b05e85686a907d7da9f41992eR191 is the same code.

You could  name your method like `formatClosingBracketNewLine` which will return a boolean.